Add `How to edit documentation` to contribution documentation guide

otegami opened the pull request:
Related Issue: https://github.com/groonga/groonga/issues/1733

I will add a new sections to the `introduction.rst` about the documentation update process.
- [x] Edit Documentation Sources <- This PR is right here!

I will do the below tasks at the following PRs.
- [ ] Reflect Your Modifications
- [ ] Preview Your Modifications on HTML
- [ ] Update Translations
- [ ] Reflect Your Translations
- [ ] Preview Your Translations on HTML
- [ ] Send Pull Request to Groonga Repository

abetomo commented on doc/locale/ja/LC_MESSAGES/contribution/documentation/introduction.po:
```diff
@@ -73,6 +73,15 @@ msgstr "以下のコマンドでHTMLファイルを生成できます。:"
+
+msgid "For Example, if you want to edit this **Introduction** page, you should edit the ``./doc/source/contribution/documentation/introduction.rst`` file."
+msgstr "例えば、この **導入** ページを編集したい場合は、``./doc/source/contribution/documentation/introduction.rst`` ファイルを編集すればよいのです。"
+
```
```
...それぞれのページに対応した ``.rst`` ファイルが必ず１つあります。
例えば、この **導入** ページを...
編集したいファイルをみつけて...
```

と、対応するファイルがある話の直後に具体例があったほうがわかりやすいと思います。

otegami replied:
確かにです！修正します。
加えて、「編集してみてください」と言及する前に具体的な事例などわかりやすい情報が出きっている方が十分な情報を持って編集に意識を向けられるので良さそうですね。

otegami replied:
fix: [3d55efd](https://github.com/groonga/groonga/pull/1734/commits/3d55efd49cbb97305a9667ab6dc198f5a8256de4) 具体例を説明の直後に持ってくる形に修正しました。

komainu8 commented on doc/locale/ja/LC_MESSAGES/contribution/documentation/introduction.po:
```diff
@@ -73,6 +73,15 @@ msgstr "以下のコマンドでHTMLファイルを生成できます。:"
+msgstr "ドキュメントを編集する方法"
+
+msgid "The Groonga documentation is written in `reStructuredText (.rst)`_. These files are located in the ``./doc/source``. Each page of the documentation corresponds to an ``.rst`` file. Please find the file you wish to edit and make your changes."
+msgstr "Groongaドキュメントは、`reStructuredText (.rst)`_ で書かれています。Groongaドキュメントのファイルは、``./doc/source`` 配下にあります。それぞれのページに対応した ``.rst`` ファイルが必ず１つあります。編集したいファイルをみつけて、編集してみてください。"
```
Markdownもあるので、Markdownについても触れているといいかと思いました。

```suggestion
msgstr "Groongaドキュメントは、`reStructuredText (.rst)`_  または Markdown (.md) で書かれています。Groongaドキュメントのファイルは、``./doc/source`` 配下にあります。それぞれのページに対応した ``.rst`` ファイルか ``.md`` ファイルが必ず１つあります。編集したいファイルをみつけて、編集してください。"
```

otegami replied:
ありがとうございます。Markdown の存在を失念していました。追加します！

otegami replied:
fix: [103ad0d](https://github.com/groonga/groonga/pull/1734/commits/103ad0d127c37eabaf6106b544d68c9bf6735e95) Markdown を利用してる旨を追記しました！

komainu8 commented on doc/locale/ja/LC_MESSAGES/contribution/documentation/introduction.po:
```diff
@@ -73,6 +73,15 @@ msgstr "以下のコマンドでHTMLファイルを生成できます。:"
+msgstr "Groongaドキュメントは、`reStructuredText (.rst)`_ で書かれています。Groongaドキュメントのファイルは、``./doc/source`` 配下にあります。それぞれのページに対応した ``.rst`` ファイルが必ず１つあります。編集したいファイルをみつけて、編集してみてください。"
+
+msgid "For Example, if you want to edit this **Introduction** page, you should edit the ``./doc/source/contribution/documentation/introduction.rst`` file."
+msgstr "例えば、この **導入** ページを編集したい場合は、``./doc/source/contribution/documentation/introduction.rst`` ファイルを編集すればよいのです。"
```
```suggestion
msgstr "例えば、この **導入** ページを編集したい場合は、``./doc/source/contribution/documentation/introduction.rst`` ファイルを編集すればよいです。"
```

otegami replied:
fix: [f70efa1](https://github.com/groonga/groonga/pull/1734/commits/f70efa1d8e7bd2765bca5b996190efef8c4ea553) 日本語がおかしい部分をまとめてこちらで修正を行いました。

komainu8 reviewed (commented):
@otegami 
I finished review. Could you confirm comments?

abetomo reviewed (approved):
I think the documentation on how to find the files is nice and helpful.

otegami commented:
@abetomo @komainu8 
レビューありがとうございます。ここまでいただいたレビューコメントの対応を行いました :pray:

komainu8 commented on doc/locale/ja/LC_MESSAGES/contribution/documentation/introduction.po:
```diff
@@ -73,6 +73,12 @@ msgstr "以下のコマンドでHTMLファイルを生成できます。:"
+msgstr "ドキュメントを編集する方法"
+
+msgid "The Groonga documentation is written in `reStructuredText (.rst)`_ or `Markdown (.md)`_ . These files are located in the ``./doc/source``. Each page of the documentation corresponds to a ``.rst`` file or a ``.md`` file. For Example, if you want to edit this **Introduction** page, you should edit the ``./doc/source/contribution/documentation/introduction.rst`` file. Please find the file you ...
+msgstr "Groongaドキュメントは、`reStructuredText (.rst)`_ または `Markdown`_ で書かれています。Groongaドキュメントのファイルは、``./doc/source`` 配下にあります。それぞれのページに対応した ``.rst`` ファイルまたは、 ``.md`` が必ず１つあります。例えば、この **導入** ページを編集したい場合は、``./doc/source/contribution/documentation/introduction.rst`` ファイルを編集すればよいです。編集したいファイルをみつけて、編集してください。"
```
```suggestion
msgstr "Groongaドキュメントは、`reStructuredText (.rst)`_ または `Markdown (.md)`_ で書かれています。Groongaドキュメントのファイルは、``./doc/source`` 配下にあります。それぞれのページに対応した ``.rst`` ファイルまたは、 ``.md`` が必ず１つあります。例えば、この **導入** ページを編集したい場合は、``./doc/source/contribution/documentation/introduction.rst`` ファイルを編集すればよいです。編集したいファイルをみつけて、編集してください。"
```

otegami replied:
fix: [2b9d89c](https://github.com/groonga/groonga/pull/1734/commits/2b9d89c33768c763c59342ea80b3f0e4168590f1)

komainu8 commented on doc/source/contribution/documentation/introduction.rst:
```diff
@@ -79,6 +79,16 @@ You can generate HTML by the following command:
+Please find the file you wish to edit and make your changes.
+
+.. _reStructuredText (.rst): https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html
+.. _Markdown (.md): https://daringfireball.net/projects/markdown/
```
@kou すみません。Groongaのドキュメントに使えるMarkdownの仕様（？）は、ここで書かれているURLで正しいでしょうか？
このリンク先が適切かどうか判断できなかったので、質問した次第です。

kou replied:
https://myst-parser.readthedocs.io/en/latest/ の方がいいかな。

Markdownは亜種がいっぱいなので実装の方にリンクしたほうがよさそう。

otegami replied:
Sphinx の Markdown ページから取ってきていました。次から取得してきたソースもコメントで添えるようにします。
- https://www.sphinx-doc.org/en/master/usage/markdown.html

> https://myst-parser.readthedocs.io/en/latest/ の方がいいかな。

なるほどです。
Sphinx も [MyST-Parser](https://myst-parser.readthedocs.io/en/latest/) を利用しているので、そちらのリンクに変更します！
- ref: https://github.com/groonga/groonga/blob/129aae9dc7c6def1be87ab78d4238797f1565416/doc/source/conf.py#L27-L31

otegami replied:
Markdown の亜種がたくさんいても対応するパーサーが扱えるものだけを見れがばよいということですね。

otegami replied:
fix: [c4393ac](https://github.com/groonga/groonga/pull/1734/commits/c4393ac9a295316e2fcc7e94a5bef0defb9e3419)

kou replied:
https://www.sphinx-doc.org/en/master/usage/markdown.html のリンクもあってもいいと思いますけど、syntaxの詳細はMyST-Parserの方にあるのでMyST-Parserのリンクはキープのほうがいいと思います。

otegami replied:
複数リンクがあってもユーザーが混乱してしまいそうなので、今回は、MyST-Parser へのリンクのみを貼る方針で行きたいと思います。

otegami commented:
@komainu8 @kou 
Thank you for reviewing it. I've just fixed what you pointed out.

otegami commented:
fix: [ff3b374](https://github.com/groonga/groonga/pull/1734/commits/ff3b37423df3b42863c0623f6f1df25e5879ef0d) I added the additional content because I think the flow of sentences isn't smooth.

---

Thank you for reviewing it. I've just fixed what you pointed out. :pray:

kou commented on doc/source/contribution/documentation/introduction.rst:
```diff
@@ -79,6 +79,17 @@ You can generate HTML by the following command:
+How to edit documentation
+--------------------------
+
+The Groonga documentation is written in `reStructuredText (.rst)`_ or `Markdown (.md)`_ . These files are located in the ``./doc/source``.
```
```suggestion
The Groonga documentation is written in `reStructuredText (.rst)`_ or `Markdown (.md)`_ . These files are located in the ``doc/source``.
```

otegami replied:
fix: [24da24f](https://github.com/groonga/groonga/pull/1734/commits/24da24fa63ff7f0a7f60ad1afa8324199cd31f85)

kou commented on doc/source/contribution/documentation/introduction.rst:
```diff
@@ -79,6 +79,17 @@ You can generate HTML by the following command:
+
+The Groonga documentation is written in `reStructuredText (.rst)`_ or `Markdown (.md)`_ . These files are located in the ``./doc/source``.
+Each page of the documentation corresponds to a ``.rst`` file or a ``.md`` file. By modifying the corresponding file, you can edit the target document.
+For Example, if you want to edit this **Introduction** page, you should edit the ``./doc/source/contribution/documentation/introduction.rst`` file.
```
```suggestion
For Example, if you want to edit this **Introduction** page, you should edit the ``doc/source/contribution/documentation/introduction.rst`` file.
```

kou commented on doc/source/contribution/documentation/introduction.rst:
```diff
@@ -79,6 +79,17 @@ You can generate HTML by the following command:
+
+The Groonga documentation is written in `reStructuredText (.rst)`_ or `Markdown (.md)`_ . These files are located in the ``./doc/source``.
+Each page of the documentation corresponds to a ``.rst`` file or a ``.md`` file. By modifying the corresponding file, you can edit the target document.
+For Example, if you want to edit this **Introduction** page, you should edit the ``./doc/source/contribution/documentation/introduction.rst`` file.
```
タイトルが変わったら自動で反映されるので`:doc:`...`にしちゃってもいいかも。

```suggestion
For Example, if you want to edit this :doc:`introduction.rst` page, you should edit the ``./doc/source/contribution/documentation/introduction.rst`` file.
```

otegami replied:
fix: [f23822b](https://github.com/groonga/groonga/pull/1734/commits/f23822b81d2e3598fbba21e3c28faade3a5fb36e) 同じページ内でも `i18n` へのリンクで利用していましたねmm
`:doc:` を利用して同ページ内にクロスリンクを張るようにしました。
- https://www.sphinx-doc.org/ja/master/usage/referencing.html#role-doc

otegami commented:
Thank you for reviewing it. I've just fixed what you pointed out. 🙏

kou commented on doc/source/contribution/documentation/introduction.rst:
```diff
@@ -79,6 +79,17 @@ You can generate HTML by the following command:
 You can find generated HTML documentation at ``../groonga.doc/doc/en/html/``.
 
+How to edit documentation
+--------------------------
```
```suggestion
-------------------------
```
